js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/VirtualizedAssetPartitionListForDialog.tsx

```diff
@@ -0,0 +1,146 @@
+        rows.push(...assetRows);
+      }
+    });
+    return rows;
+  }, [assetKeysByPartition, expandedPartitions]);
+
+  const rowVirtualizer = useVirtualizer({
+    count: allRows.length,
+    getScrollElement: () => container.current,
+    estimateSize: () => 40,
+    overscan: 10,
+  });
+
+  const onToggle = React.useCallback((partitionName: string) => {
+    setExpandedPartitions((current) => {
+      const copy = new Set(Array.from(current));
+      if (current.has(partitionName)) {
+        copy.delete(partitionName);
+      } else {
+        copy.add(partitionName);
+      }
+      return copy;
+    });
+  }, []);
+
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  return (
+    <Container ref={container} style={{padding: '8px 24px'}}>
+      <Inner $totalHeight={totalHeight}>
+        {items.map(({index, key, size, start}) => {
+          const row = allRows[index]!;
+          return (
+            <Row $height={size} $start={start} key={key}>
+              <Box
+                style={{height: '100%'}}
+                flex={{direction: 'row', alignItems: 'center'}}
+                border={
+                  index < allRows.length - 1
+                    ? {side: 'bottom', width: 1, color: Colors.KeylineGray}
+                    : null
+                }
+              >
+                {row.type === 'partition-name' ? (
+                  <ExpandablePartitionName
+                    partitionName={row.partitionName}
+                    expanded={row.expanded}
+                    detail={renderPartitionDetail(row)}
+                    onToggle={onToggle}
+                  />
+                ) : (
+                  <Box padding={{left: 24}}>
+                    {renderItem(row.assetKey)}
+                    {/* <AssetLink path={row.assetKey.path} icon="asset" /> */}
+                  </Box>
+                )}
+              </Box>
+            </Row>
+          );
+        })}
+      </Inner>
+    </Container>
+  );
+}
+
+interface ExpandablePartitionNameProps {
+  partitionName: string;
+  expanded: boolean;
+  detail: React.ReactNode;
+  onToggle: (partitionName: string) => void;
+}
+
+const ExpandablePartitionName = ({
+  partitionName,
+  detail,
+  expanded,
+  onToggle,
+}: ExpandablePartitionNameProps) => {
+  return (
+    <PartitionNameButton onClick={() => onToggle(partitionName)}>
+      <Icon
+        name="arrow_drop_down"
+        style={{transform: expanded ? 'rotate(0deg)' : 'rotate(-90deg)'}}
+      />
+      <div>{partitionName}</div>
+      <div>{detail}</div>
+    </PartitionNameButton>
+  );
+};
+
+const PartitionNameButton = styled.button`
+  background-color: transparent;
+  cursor: pointer;
+  padding: 0;
+  border: 0;
+  display: flex;
+  height: 100%;
+  width: 100%;
+  flex-direction: row;
+  align-items: center;
+  gap: 8px;
+
+  :focus {
+    outline: none;
+  }
+`;
```

Comment:
Generic virtualized list for showing a list of partitioned assets (grouped by partition name, collapsible, etc) in one of these dialogs.